danbooru.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2019 Nick Filmer (nick80835@gmail.com)
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import requests
+
+from .. import loader, utils
+
+DAN_URL = "http://danbooru.donmai.us/posts.json"
+
+
+def register(cb):
+    cb(DanbooruMod())
+
+
+class DanbooruMod(loader.Module):
+    """Danbooru module for scraping Danbooru"""
+    def __init__(self):
+        self.name = _("Danbooru")
+
+    async def dancmd(self, message):
+        """Non-explicit scraping"""
+        search_query = message.message.lstrip("dan")
+        await self._runscrape(message, search_query, "Safe")
+
+    async def danxcmd(self, message):
```

Comment:
Please make this a separate module. I won't put explicit stuff in the official repo.